src/wallet/KeyChain.ts

```diff
@@ -51,6 +144,27 @@ export class KeyChain {
 		this.getCheapestKeyset();
 	}
 
+	/**
+	 * Synchronously load keysets and keys from cached data.
+	 *
+	 * @remarks
+	 * Does not hit the network. Intended for callers that already have a KeyChainCache and want a
+	 * synchronous path.
+	 */
+	loadFromCache(cache: KeyChainCache): void {
+		if (cache.unit !== this.unit) {
+			throw new Error(
+				`KeyChain unit mismatch in cache, expected '${this.unit}', got '${cache.unit}'`,
```

Comment:
When the cache unit doesn't match, the error message could be more helpful by including the mintUrl from the cache to help developers debug where the mismatched cache came from. Its a nit, but could really prevent a headache for devs. something small along the lines of:

throw new Error(
    `KeyChain unit mismatch in cache, expected '${this.unit}', got '${cache.unit}' (from ${cache.mintUrl})`
);